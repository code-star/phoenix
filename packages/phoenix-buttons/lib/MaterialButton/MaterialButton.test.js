import * as renderer from 'react-test-renderer'
import {
  // contained
  containedDefaultMaterialButton,
  containedPrimaryMaterialButton,
  containedSecondaryMaterialButton,
  containedDisabledMaterialButton,
  containedLinkMaterialButton,
  containedUploadMaterialButton,
  // text
  textDefaultMaterialButton,
  textPrimaryMaterialButton,
  textSecondaryMaterialButton,
  textDisabledMaterialButton,
  textLinkMaterialButton,
  textUploadMaterialButton,
  // outlined
  outlinedDefaultMaterialButton,
  outlinedPrimaryMaterialButton,
  outlinedSecondaryMaterialButton,
  outlinedDisabledMaterialButton,
  outlinedLinkMaterialButton,
  outlinedUploadMaterialButton,
} from './MaterialButton.render'

describe('<MaterialButton />', () => {
  describe('Snaphot', () => {
    describe('contained', () => {
      test('must match containedDefaultMaterialButton', () => {
        expect(renderer.create(containedDefaultMaterialButton).toJSON()).toMatchSnapshot()
      })

      test('must match containedPrimaryMaterialButton', () => {
        expect(renderer.create(containedPrimaryMaterialButton).toJSON()).toMatchSnapshot()
      })

      test('must match containedSecondaryMaterialButton', () => {
        expect(renderer.create(containedSecondaryMaterialButton).toJSON()).toMatchSnapshot()
      })

      test('must match containedDisabledMaterialButton', () => {
        expect(renderer.create(containedDisabledMaterialButton).toJSON()).toMatchSnapshot()
      })

      test('must match containedLinkMaterialButton', () => {
        expect(renderer.create(containedLinkMaterialButton).toJSON()).toMatchSnapshot()
      })

      test('must match containedUploadMaterialButton', () => {
        expect(renderer.create(containedUploadMaterialButton).toJSON()).toMatchSnapshot()
      })
    })

    describe('text', () => {
      test('must match textDefaultMaterialButton', () => {
        expect(renderer.create(textDefaultMaterialButton).toJSON()).toMatchSnapshot()
      })

      test('must match textPrimaryMaterialButton', () => {
        expect(renderer.create(textPrimaryMaterialButton).toJSON()).toMatchSnapshot()
      })

      test('must match textSecondaryMaterialButton', () => {
        expect(renderer.create(textSecondaryMaterialButton).toJSON()).toMatchSnapshot()
      })

      test('must match textDisabledMaterialButton', () => {
        expect(renderer.create(textDisabledMaterialButton).toJSON()).toMatchSnapshot()
      })

      test('must match textLinkMaterialButton', () => {
        expect(renderer.create(textLinkMaterialButton).toJSON()).toMatchSnapshot()
      })

      test('must match textUploadMaterialButton', () => {
        expect(renderer.create(textUploadMaterialButton).toJSON()).toMatchSnapshot()
      })
    })

    describe('outlined', () => {
      test('must match outlinedDefaultMaterialButton', () => {
        expect(renderer.create(outlinedDefaultMaterialButton).toJSON()).toMatchSnapshot()
      })

      test('must match outlinedPrimaryMaterialButton', () => {
        expect(renderer.create(outlinedPrimaryMaterialButton).toJSON()).toMatchSnapshot()
      })

      test('must match outlinedSecondaryMaterialButton', () => {
        expect(renderer.create(outlinedSecondaryMaterialButton).toJSON()).toMatchSnapshot()
      })

      test('must match outlinedDisabledMaterialButton', () => {
        expect(renderer.create(outlinedDisabledMaterialButton).toJSON()).toMatchSnapshot()
      })

      test('must match outlinedLinkMaterialButton', () => {
        expect(renderer.create(outlinedLinkMaterialButton).toJSON()).toMatchSnapshot()
      })

      test('must match outlinedUploadMaterialButton', () => {
        expect(renderer.create(outlinedUploadMaterialButton).toJSON()).toMatchSnapshot()
      })
    })
  })
})
