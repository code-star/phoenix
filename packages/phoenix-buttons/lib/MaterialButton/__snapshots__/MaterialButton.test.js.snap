// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MaterialButton /> Snaphot contained must match containedDefaultMaterialButton 1`] = `
<button
  className="MuiButtonBase-root-29 MuiButton-root-3 MuiButton-contained-14 MuiButton-raised-17 MuiButton-colorInherit-24 MaterialButton-button-1"
  disabled={false}
  href=""
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex="0"
  type="button"
>
  <span
    className="MuiButton-label-4"
  >
    Default
  </span>
  <span
    className="MuiTouchRipple-root-32"
  />
</button>
`;

exports[`<MaterialButton /> Snaphot contained must match containedDisabledMaterialButton 1`] = `
<button
  className="MuiButtonBase-root-29 MuiButtonBase-disabled-30 MuiButton-root-3 MuiButton-contained-14 MuiButton-raised-17 MuiButton-disabled-23 MuiButton-colorInherit-24 MaterialButton-button-1"
  disabled={true}
  href=""
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex="-1"
  type="button"
>
  <span
    className="MuiButton-label-4"
  >
    Disabled
  </span>
</button>
`;

exports[`<MaterialButton /> Snaphot contained must match containedLinkMaterialButton 1`] = `
<a
  className="MuiButtonBase-root-29 MuiButton-root-3 MuiButton-contained-14 MuiButton-raised-17 MuiButton-colorInherit-24 MaterialButton-button-1"
  href="https://github.com/code-star/phoenix"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  role="button"
  tabIndex="0"
>
  <span
    className="MuiButton-label-4"
  >
    Link
  </span>
  <span
    className="MuiTouchRipple-root-32"
  />
</a>
`;

exports[`<MaterialButton /> Snaphot contained must match containedPrimaryMaterialButton 1`] = `
<button
  className="MuiButtonBase-root-29 MuiButton-root-3 MuiButton-contained-14 MuiButton-containedPrimary-15 MuiButton-raised-17 MuiButton-raisedPrimary-18 MaterialButton-button-1"
  disabled={false}
  href=""
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex="0"
  type="button"
>
  <span
    className="MuiButton-label-4"
  >
    Primary
  </span>
  <span
    className="MuiTouchRipple-root-32"
  />
</button>
`;

exports[`<MaterialButton /> Snaphot contained must match containedSecondaryMaterialButton 1`] = `
<button
  className="MuiButtonBase-root-29 MuiButton-root-3 MuiButton-contained-14 MuiButton-containedSecondary-16 MuiButton-raised-17 MuiButton-raisedSecondary-19 MaterialButton-button-1"
  disabled={false}
  href=""
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex="0"
  type="button"
>
  <span
    className="MuiButton-label-4"
  >
    Secondary
  </span>
  <span
    className="MuiTouchRipple-root-32"
  />
</button>
`;

exports[`<MaterialButton /> Snaphot contained must match containedUploadMaterialButton 1`] = `
Array [
  <input
    accept="image/*"
    className="MaterialButton-input-2"
    id="contained-button-file"
    multiple={true}
    type="file"
  />,
  <label
    htmlFor="contained-button-file"
  >
    <span
      className="MuiButtonBase-root-29 MuiButton-root-3 MuiButton-contained-14 MuiButton-raised-17 MaterialButton-button-1"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex="0"
    >
      <span
        className="MuiButton-label-4"
      >
        Upload
      </span>
      <span
        className="MuiTouchRipple-root-32"
      />
    </span>
  </label>,
]
`;

exports[`<MaterialButton /> Snaphot outlined must match outlinedDefaultMaterialButton 1`] = `
<button
  className="MuiButtonBase-root-29 MuiButton-root-3 MuiButton-outlined-11 MuiButton-colorInherit-24 MaterialButton-button-1"
  disabled={false}
  href=""
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex="0"
  type="button"
>
  <span
    className="MuiButton-label-4"
  >
    Default
  </span>
  <span
    className="MuiTouchRipple-root-32"
  />
</button>
`;

exports[`<MaterialButton /> Snaphot outlined must match outlinedDisabledMaterialButton 1`] = `
<button
  className="MuiButtonBase-root-29 MuiButtonBase-disabled-30 MuiButton-root-3 MuiButton-outlined-11 MuiButton-disabled-23 MuiButton-colorInherit-24 MaterialButton-button-1"
  disabled={true}
  href=""
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex="-1"
  type="button"
>
  <span
    className="MuiButton-label-4"
  >
    Disabled
  </span>
</button>
`;

exports[`<MaterialButton /> Snaphot outlined must match outlinedLinkMaterialButton 1`] = `
<a
  className="MuiButtonBase-root-29 MuiButton-root-3 MuiButton-outlined-11 MuiButton-colorInherit-24 MaterialButton-button-1"
  href="https://github.com/code-star/phoenix"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  role="button"
  tabIndex="0"
>
  <span
    className="MuiButton-label-4"
  >
    Link
  </span>
  <span
    className="MuiTouchRipple-root-32"
  />
</a>
`;

exports[`<MaterialButton /> Snaphot outlined must match outlinedPrimaryMaterialButton 1`] = `
<button
  className="MuiButtonBase-root-29 MuiButton-root-3 MuiButton-outlined-11 MuiButton-outlinedPrimary-12 MaterialButton-button-1"
  disabled={false}
  href=""
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex="0"
  type="button"
>
  <span
    className="MuiButton-label-4"
  >
    Primary
  </span>
  <span
    className="MuiTouchRipple-root-32"
  />
</button>
`;

exports[`<MaterialButton /> Snaphot outlined must match outlinedSecondaryMaterialButton 1`] = `
<button
  className="MuiButtonBase-root-29 MuiButton-root-3 MuiButton-outlined-11 MuiButton-outlinedSecondary-13 MaterialButton-button-1"
  disabled={false}
  href=""
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex="0"
  type="button"
>
  <span
    className="MuiButton-label-4"
  >
    Secondary
  </span>
  <span
    className="MuiTouchRipple-root-32"
  />
</button>
`;

exports[`<MaterialButton /> Snaphot outlined must match outlinedUploadMaterialButton 1`] = `
Array [
  <input
    accept="image/*"
    className="MaterialButton-input-2"
    id="outlined-button-file"
    multiple={true}
    type="file"
  />,
  <label
    htmlFor="outlined-button-file"
  >
    <span
      className="MuiButtonBase-root-29 MuiButton-root-3 MuiButton-outlined-11 MaterialButton-button-1"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex="0"
    >
      <span
        className="MuiButton-label-4"
      >
        Upload
      </span>
      <span
        className="MuiTouchRipple-root-32"
      />
    </span>
  </label>,
]
`;

exports[`<MaterialButton /> Snaphot text must match textDefaultMaterialButton 1`] = `
<button
  className="MuiButtonBase-root-29 MuiButton-root-3 MuiButton-text-5 MuiButton-flat-8 MuiButton-colorInherit-24 MaterialButton-button-1"
  disabled={false}
  href=""
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex="0"
  type="button"
>
  <span
    className="MuiButton-label-4"
  >
    Default
  </span>
  <span
    className="MuiTouchRipple-root-32"
  />
</button>
`;

exports[`<MaterialButton /> Snaphot text must match textDisabledMaterialButton 1`] = `
<button
  className="MuiButtonBase-root-29 MuiButtonBase-disabled-30 MuiButton-root-3 MuiButton-text-5 MuiButton-flat-8 MuiButton-disabled-23 MuiButton-colorInherit-24 MaterialButton-button-1"
  disabled={true}
  href=""
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex="-1"
  type="button"
>
  <span
    className="MuiButton-label-4"
  >
    Disabled
  </span>
</button>
`;

exports[`<MaterialButton /> Snaphot text must match textLinkMaterialButton 1`] = `
<a
  className="MuiButtonBase-root-29 MuiButton-root-3 MuiButton-text-5 MuiButton-flat-8 MuiButton-colorInherit-24 MaterialButton-button-1"
  href="https://github.com/code-star/phoenix"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  role="button"
  tabIndex="0"
>
  <span
    className="MuiButton-label-4"
  >
    Link
  </span>
  <span
    className="MuiTouchRipple-root-32"
  />
</a>
`;

exports[`<MaterialButton /> Snaphot text must match textPrimaryMaterialButton 1`] = `
<button
  className="MuiButtonBase-root-29 MuiButton-root-3 MuiButton-text-5 MuiButton-textPrimary-6 MuiButton-flat-8 MuiButton-flatPrimary-9 MaterialButton-button-1"
  disabled={false}
  href=""
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex="0"
  type="button"
>
  <span
    className="MuiButton-label-4"
  >
    Primary
  </span>
  <span
    className="MuiTouchRipple-root-32"
  />
</button>
`;

exports[`<MaterialButton /> Snaphot text must match textSecondaryMaterialButton 1`] = `
<button
  className="MuiButtonBase-root-29 MuiButton-root-3 MuiButton-text-5 MuiButton-textSecondary-7 MuiButton-flat-8 MuiButton-flatSecondary-10 MaterialButton-button-1"
  disabled={false}
  href=""
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex="0"
  type="button"
>
  <span
    className="MuiButton-label-4"
  >
    Secondary
  </span>
  <span
    className="MuiTouchRipple-root-32"
  />
</button>
`;

exports[`<MaterialButton /> Snaphot text must match textUploadMaterialButton 1`] = `
Array [
  <input
    accept="image/*"
    className="MaterialButton-input-2"
    id="flat-button-file"
    multiple={true}
    type="file"
  />,
  <label
    htmlFor="flat-button-file"
  >
    <span
      className="MuiButtonBase-root-29 MuiButton-root-3 MuiButton-text-5 MuiButton-flat-8 MaterialButton-button-1"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex="0"
    >
      <span
        className="MuiButton-label-4"
      >
        Upload
      </span>
      <span
        className="MuiTouchRipple-root-32"
      />
    </span>
  </label>,
]
`;
